app/models/katello/content_view_version.rb

```diff
@@ -247,8 +247,10 @@ def package_groups
       PackageGroup.in_repositories(archived_repos).uniq
     end
 
-    def check_ready_to_promote!
+    def check_ready_to_promote!(to_env)
       fail _("Default content view versions cannot be promoted") if default?
+      content_view.check_composite_action_allowed!(to_env)
+      true
```

Comment:
this true could be considered unnecessary, as you could allow the return value of check_composite_action_allowed! to dictate the return